simtools/simulator.py

```diff
@@ -313,42 +326,55 @@ def _set_simulation_runner(self):
         Set simulation runners
 
         """
+        common_args = {
+            "label": self.label,
+            "simtel_source_path": self._simulator_source_path,
+        }
+        corsika_args = {
+            "mongo_db_config": self._mongo_db_config,
+            "site": self.site,
+            "layout_name": self.layout_name,
+            "corsika_parameters_file": self._corsika_parameters_file,
+            "corsika_config_data": self._corsika_config_data,
+        }
+        # TODO: This is not very elegant, find a nicer solution?
+        if self.simulator in ["simtel", "corsika_simtel"]:
+            simtel_args = {
+                "array_model": self.array_model,
+                "config_data": {
+                    "simtel_data_directory": self.config.data_directory,
+                    "primary": self.config.primary,
+                    "zenith_angle": self.config.zenith_angle * u.deg,
+                    "azimuth_angle": self.config.azimuth_angle * u.deg,
+                },
+            }
+        if self.simulator == "corsika":
+            self._set_corsika_runner(common_args | corsika_args)
         if self.simulator == "simtel":
-            self._set_simtel_runner()
-        elif self.simulator == "corsika":
-            self._set_corsika_runner()
+            self._set_simtel_runner(common_args | simtel_args)
+        if self.simulator == "corsika_simtel":
+            self._set_corsika_simtel_runner(common_args, corsika_args, simtel_args)
 
```

Comment:
Dont we need a final 'else' here to check for possible invalid `self.simulator`?

Reply:
No, cause it is checked already in the setter:
https://github.com/gammasim/gammasim-tools/blob/4a1b76426f5fb852e48d453b089c42bb955f265d/simtools/simulator.py#L159

However, this comment made me realise that this is not a proper setter so I will change that later.